validator/sql_location_test.rego

```diff
@@ -0,0 +1,165 @@
+#
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+package templates.gcp.GCPSQLLocationConstraintV1
+
+import data.test.fixtures.sql_location.assets.sql_instances as fixture_sql_instances
+import data.test.fixtures.sql_location.constraints as fixture_constraints
+
+find_violations[violation] {
+	asset := data.assets[_]
+	constraint := data.test_constraints[_]
+	issues := deny with input.asset as asset with input.constraint as constraint
+	total_issues := count(issues)
+	violation := issues[_]
+}
+
+# Test logic for allowlisting/denylisting
+test_target_location_match_count_allowlist {
+	target_location_match_count("allowlist", match_count)
+	match_count == 0
+}
+
+test_target_location_match_count_denylist {
+	target_location_match_count("denylist", match_count)
+	match_count == 1
+}
+
+# Test for no violations with empty parameters
+violations_with_empty_parameters[violation] {
+	constraints := [fixture_constraints.location_default]
+	found_violations := find_violations with data.assets as fixture_sql_instances
+		 with data.test_constraints as constraints
+
+	violation := found_violations[_]
+}
+
+test_sql_instance_default {
+	found_violations := violations_with_empty_parameters
+	count(found_violations) == 0
+}
+
+# Test empty denylist
+violations_with_empty_denylist[violation] {
+	constraints := [fixture_constraints.denylist_none]
+	found_violations := find_violations with data.assets as fixture_sql_instances
+		 with data.test_constraints as constraints
+
+	violation := found_violations[_]
+}
+
+test_sql_instance_denylist_none {
+	found_violations := violations_with_empty_denylist
+	count(found_violations) == 0
+}
+
+# Test empty allowlist
+violations_with_empty_allowlist[violation] {
+	constraints := [fixture_constraints.allowlist_none]
+	found_violations := find_violations with data.assets as fixture_sql_instances
+		 with data.test_constraints as constraints
+
+	violation := found_violations[_]
+}
+
+test_sql_instance_allowlist_none {
+	found_violations := violations_with_empty_allowlist
+	count(found_violations) == count(fixture_sql_instances)
```

Comment:
instead of count, can we compare the instance names to be more precise? Similarly below.

Reply:
added test against instance names